packages/element-path/src/element-path.ts

```diff
@@ -409,27 +434,55 @@ export class ElementPath {
 
     public generateChildElementsPath(key: string | number): ElementPath {
         if (isInteger(key)) {
-            if (hasOwn(this.searchOptions, 'index')) {
-                throw Error('Can not select index element from already sliced element');
-            }
+            return this.generateChildElementPathByOptions(Object.assign({}, this.searchOptions, {
+                index: +key
```

Comment:
why `+` is added (`+key`)?

consider use Math.abs, if you need only positive numbers. in case `key < 0`, index still be `< 0`